salt/minion.py

```diff
@@ -2624,7 +2624,7 @@ def tune_in(self, start=True):
             def ping_master():
                 try:
                     def ping_timeout_handler(*_):
-                        if not self.opts.get('auth_safemode', True):
+                        if self.opts.get('auth_safemode', True):
```

Comment:
Looks like this was implemented incorrectly, but we should not have it switch to True, if it is not specified, it should be `False` in the .get() as well.

Reply:
@rongzeng54 Did you see the feedback here?